### PR TITLE
Don't filter autocomplete suggestions by available subjects

### PR DESCRIPTION
**Pull request by hancush**
## Overview

This PR tweaks the previously approved SmartLogic refactor so that autocomplete suggestions are not filtered by available terms, which is the current behavior of the site.

## Testing Instructions

 * Confirm CI passes!
 * Confirm the code changes look good.

### CODE REVIEW

File: lametro/api.py [hunk 1]
```diff
@@ -110,24 +110,45 @@ def get_relations(self, concepts):
...
+                    "display_name": base_obj["name"] + concept.get("synonym_label", ""),
+                    "guid": guid,
+                }
             )
 
         return {
             "status_code": 200,
             "concepts": concepts,
-            "subjects": subjects,
+            "subjects": subjects[:result_count],
```
Comment (xmedr) [inline lametro/api.py:151]: What is this bracket/colon combo doing? It was a bit hard looking this one up.

**Review by xmedr — APPROVED** [review]

File: lametro/api.py [hunk 2]
```diff
@@ -110,24 +110,45 @@ def get_relations(self, concepts):
...
+            # continue if there is not a corresponding subject
+            if filter_concepts:
+                base_obj = matched_subjects.get(guid)
+                if not base_obj:
+                    continue
+
+            # Otherwise, fall back to the concept as it is represented
+            # in the SES API
+            else:
+                base_obj = matched_subjects.get(guid, concept)
```
[the remark this answers is not in the document]
Comment (hancush) [reply lametro/api.py:138]: Yes, it's so nice! I came to dict.get so late. Big fan, lol.

File: lametro/api.py (hunk 1 above)
Comment (hancush) [reply lametro/api.py:151]: This is called [slice notation](https://realpython.com/lessons/indexing-and-slicing/)! Here, it returns up to the requested number of results from `subjects`.

Comment (xmedr) [reply lametro/api.py:151]: Ooooh gotcha, ty! For some reason I had thought that `subjects` here was a dict, not a list. This makes a lot of sense.

**Pull request merged**